### PR TITLE
fix(code-highlight): highlight python function and method call sites

### PR DESCRIPTION
**Pull request by hwkr**
## Problem

Python code samples looked mostly grey and unstyled next to other languages. Turns out highlight.js's Python grammar only colors `def`/`class` definitions and a fixed list of built-ins — it never colors call sites. Since our API snippets are basically all method calls (`conn.request(...)`, `response.json()`), almost everything fell back to the default grey text.

## Solution

* New custom Python grammar (`languages/python.ts`) that extends the bundled one to color function and method call sites as function titles — same trick highlight.js's own JavaScript grammar uses (word-boundary anchor + negative lookahead so keywords and built-ins keep their scopes)
* Also removed a dead duplicate `.hljs-built_in` rule in `code.css` — leftover from the Prism→highlight.js migration, it sat outside `.scalar-app` so it was already being overridden anyway

Before / After

CleanShot 2026-06-11 at 20 16 34@2x

CleanShot 2026-06-11 at 20 16 43@2x

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset.
- [x] I added tests.
- [ ] I updated the documentation.

## Ticket

Fixes DOC-5618

---

> [!NOTE]
> **Low Risk**
> Presentation-only syntax highlighting in `@scalar/code-highlight`; no auth, data, or runtime API changes, with regression tests for grammar edge cases.
> 
> **Overview**
> Python API snippets were mostly default grey because the bundled highlight.js grammar never scopes **call sites**—only `def`/`class` names and built-ins. This PR **replaces** the stock Python language with a wrapper in `languages/python.ts` that adds one rule: identifiers immediately before `(` get `title.function` (orange), with keywords/built-ins/literals/types excluded via negative lookahead and word boundaries (including a performance-conscious lookahead design for untrusted input).
> 
> **`basic.ts`** and **`standard.ts`** now import `./python` instead of `highlight.js/lib/languages/python`. **`code.css`** drops a redundant global `.hljs-built_in` block that duplicated styling already defined under `.scalar-app`. Vitest coverage in **`python.test.ts`** locks in call-site highlighting, keyword/`if (` edge cases, built-ins, and defs/classes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 231e21102dd93a40b6e05a8083d1af9ee39fa37f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/code-highlight/src/css/code.css [hunk 1]
```diff
@@ -188,7 +188,3 @@
 .hljs.language-objectivec .hljs-built_in {
   color: var(--scalar-color-orange);
 }
-
-.hljs-built_in {
```
Comment (hwkr) [inline packages/code-highlight/src/css/code.css:192]: Duplicate rule

**Review by hanspagel — APPROVED** [review]
niice
